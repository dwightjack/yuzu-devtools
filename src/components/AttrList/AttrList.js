import { html } from 'lit-html';
import * as styles from './AttrList.styles';

const ATTRS = ['ref', 'detached'];

export default function AttrList(props) {
  return ATTRS.map((attr) => {
    const val = props[attr];
    const bool = typeof val === 'boolean' ? null : val;
    return val
      ? html`
          <span class="${styles.root}" data-value="${bool}"
            >${attr + (bool ? '=' : '')}</span
          >
        `
      : null;
  }).filter((x) => x);
}

AttrList.ATTRS = ATTRS;
