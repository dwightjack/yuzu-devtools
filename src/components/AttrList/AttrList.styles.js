import { css } from 'emotion';

export const root = css`
  margin: 0 0.5em;
  color: var(--color-accent-ter);

  &[data-value]::after {
    content: '"' attr(data-value) '"';
    color: var(--color-accent);
  }
`;
