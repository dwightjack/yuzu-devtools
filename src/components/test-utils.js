export const mockComponent = (tag) => (props) => {
  const el = document.createElement(tag);
  Object.keys(props).forEach((key) => {
    el.setAttribute(key, props[key]);
  });
  return el;
};
