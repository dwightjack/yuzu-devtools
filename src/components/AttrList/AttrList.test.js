import { toHTML } from '../test-utils';
import AttrList from './AttrList';

describe('AttrList', () => {
  test('matches default snapshot', () => {
    const { html } = toHTML(AttrList, {
      ref: 'DEMO',
      detached: true,
    });
    expect(html).toMatchSnapshot();
  });

  test('matches empty snapshot', () => {
    const { html } = toHTML(AttrList, {});
    expect(html).toMatchSnapshot();
  });
});
