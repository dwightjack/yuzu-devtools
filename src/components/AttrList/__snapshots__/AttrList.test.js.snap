// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttrList matches default snapshot 1`] = `
<span>
  ref
  =
  <b>
    "DEMO"
  </b>
</span>
<span>
  detached
</span>
`;

exports[`AttrList matches empty snapshot 1`] = ``;
